File: go/pangram/pangram.go
// Package pangram tests for pangrams.
package pangram

import "strings"

func IsPangram(s string) bool {
	s = strings.ToLower(s)
	for i := 'a'; i <= 'z'; i++ {
		if ! strings.ContainsRune(s, i) {
			return false
		}
	}
	return true
}

File: haskell/pangram/src/Pangram.hs
module Pangram (isPangram) where

import qualified Data.Char as C
import qualified Data.Set as S


isPangram :: String -> Bool
isPangram text = asciiLower `S.isSubsetOf` input
  where
    input = S.fromList $ map C.toLower text
    asciiLower = S.fromList ['a'..'z']

-- vim:ts=2:sw=2:expandtab
